internal/proto/processor/v1/processor.proto

```diff
@@ -0,0 +1,146 @@
+syntax = "proto3";
```

Comment:
I'd maybe also move `proto` into the `wasm` package, since that should be the only place it's used, right? And if we then expose `wasm` (as I proposed in the other comment) we will have access to these proto structs in Conduit, which means we don't need to separately expose `serde`. Less abstraction, less indirection.